activerecord/lib/active_record/timestamp.rb

```diff
@@ -159,7 +159,7 @@ def max_updated_column_timestamp
     def clear_timestamp_attributes
       all_timestamp_attributes_in_model.each do |attribute_name|
         self[attribute_name] = nil
-        clear_attribute_changes([attribute_name])
+        clear_attribute_change(attribute_name)
```

Comment:
This should probably go through the attribute method to resolve the alias, as in `increment!`. However removing this line entirely doesn't cause the test suite to fail, so I'm not sure it's even necessary at all.

Reply:
Note that `all_timestamp_attributes_in_model` array is already aliases resolved.

Reply:
Yes, you're right! Maybe we should be using `_write_attribute` instead of `[]=` on the line above then, to avoid resolving twice? That's a very strange edge case though, perhaps it's not worth covering.

Reply:
Yes, we have `[]=`, `write_attribute`, and `_write_attribute`, all `[]=` could replace to faster variant `write_attribute`, and some `write_attribute` and `[]=` could replace to fastest variant `write_attribute`, all places are in my rader.